consumer/pdatautil/metricsbuilder_test.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdatautil
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/internal/data"
+	"go.opentelemetry.io/collector/testutil/metricstestutil"
+)
+
+func ExampleMetricsBuilder() {
+	mb := MetricsBuilder{
+		InstrumentationLibraryName:    "example",
+		InstrumentationLibraryVersion: "0.1",
+		ResourceAttributes: map[string]string{
+			"host": "my-host",
+		},
+		Timestamp: time.Now(),
+	}
+
+	// All metrics added after this will have these labels
+	mb.WithLabels(map[string]string{
+		"disk": "sda",
+	}).AddGauge("disk.usage", 9000000)
+
+	// Add metrics in a chained manner
+	mb.
+		AddGauge("disk.capacity", 9000000).
+		AddCumulative("disk.reads", 50)
```

Comment:
Is this indentation dictated by go fmt? (Looks a bit strange).

Reply:
Yes, the way Github does tabs with 8 spaces also doesn't help.